xarray/core/common.py

```diff
@@ -347,6 +347,34 @@ def _ipython_key_completions_(self) -> list[str]:
         return list(items)
 
 
+class TreeAttrAccessMixin(AttrAccessMixin):
```

Comment:
So here's the biggest question I have... is this okay for now? (Maybe this is directed at @TomNicholas)

The use of `__slots__` is a little new to me (but makes sense from reading up on it). I think to adopt `__slots__` and remove `__dict__` I would need to slightly rework `DataTree`, `TreeNode` and `NamedNode`. I wanted to check that the scope mainly seems to be `self.children` and `self.parent` attributes, and understand why those classes were implemented as they currently are before trying to alter things.

Reply:
Good catch. I accidentally commented out that last line. Will fix!